packages/anon-aadhaar-react/src/components/ProveButton.tsx

```diff
@@ -1,66 +1,74 @@
 import { AnonAadhaarPCDArgs } from 'anon-aadhaar-pcd'
 import { ArgumentTypeName } from '@pcd/pcd-types'
 import styled from 'styled-components'
-import { useContext } from 'react'
+import { Dispatch, useContext, SetStateAction } from 'react'
 import { AnonAadhaarContext } from '../hooks/useAnonAadhaar'
 import { Spinner } from './LoadingSpinner'
-import { AadhaarSignatureValidition } from '../interface'
-import { APP_ID } from '../constants'
+import React from 'react'
+import { extractWitness } from 'anon-aadhaar-pcd'
 
 interface ProveButtonProps {
-  msgBigInt?: bigint
-  modulusBigInt?: bigint
-  sigBigInt?: bigint
-  signatureValidity: '' | AadhaarSignatureValidition
+  pdfData: Buffer
+  password: string
+  provingEnabled: boolean
+  setErrorMessage: Dispatch<SetStateAction<string | null>>
 }
 
 export const ProveButton: React.FC<ProveButtonProps> = ({
-  msgBigInt,
-  modulusBigInt,
-  sigBigInt,
-  signatureValidity,
+  pdfData,
+  password,
+  provingEnabled,
+  setErrorMessage,
 }) => {
-  const { state, startReq } = useContext(AnonAadhaarContext)
+  const { state, startReq, appId } = useContext(AnonAadhaarContext)
 
-  const args: AnonAadhaarPCDArgs = {
-    base_message: {
-      argumentType: ArgumentTypeName.BigInt,
-      userProvided: false,
-      value: msgBigInt?.toString(),
-      description: '',
-    },
-    signature: {
-      argumentType: ArgumentTypeName.BigInt,
-      userProvided: false,
-      value: sigBigInt?.toString(),
-      description: '',
-    },
-    modulus: {
-      argumentType: ArgumentTypeName.BigInt,
-      userProvided: false,
-      value: modulusBigInt?.toString(),
-      description: '',
-    },
-    app_id: {
-      argumentType: ArgumentTypeName.BigInt,
-      userProvided: false,
-      value: APP_ID,
-      description: '',
-    },
+  const startProving = async () => {
+    try {
+      if (appId === null) throw new Error('Missing application Id!')
```

Comment:
If appId is null then client will throw error on console log?

Reply:
Yes, and also in the error component displayed to the user.